src/components/Board.js

```diff
@@ -16,18 +26,101 @@ class Board extends Component {
     };
   }
 
+
+
+  addCard = (cardInfo) => {
+    console.log(cardInfo);
+    const url = Cards_URL + this.props.boardName + '/cards';
+    axios.post(url, cardInfo)
+    .then((response) => {
+      console.log(response.data);
+      this.props.updateStatusCallback(`Successfully added card ${ response.data.card.id}!`, 'success');
+
+      let updatedData = this.state.cards;
+      updatedData.push(response.data);
+      this.setState({cards: updatedData});
+    })
+    .catch((error) => {
+      this.props.updateStatusCallback(`Error adding card ${ cardInfo.id }.`, 'error');
+      this.setState({ error: error.message });
+    });
+  }
+
+  deleteCard = (index, id) => {
+    let updatedCards = this.state.cards;
+    updatedCards.splice(index, 1);
+    this.setState({cards: updatedCards});
+
+    let DELETE_URL = Cards_URL + this.props.boardName + '/cards/' + id;
+    axios.delete(DELETE_URL)
+    .then((response) => {
+      console.log(response);
+      this.props.updateStatusCallback('Successfully deleted card...', 'success')
+    })
+    .catch((error) => {
+      this.props.updateStatusCallback(error.message, 'error')
+    });
+  }
+
+  componentDidMount() {
+    this.getBoard(Cards_URL, this.props.boardName)
+  }
+  getBoard(url, boardName) {
+    this.props.updateStatusCallback('Loading cards...', 'success');
+    const cardUrl = Cards_URL + boardName + '/cards';
+    axios.get(cardUrl)
+    .then((response) => {
+      console.log("success");
+      console.log(response);
+
+      this.props.updateStatusCallback('Successfully load cards...', 'success')
+      const cards = response.data
+      this.setState({ cards: cards });
+    })
+    .catch((error) => {
+      console.log("Error: (')");
+      console.log("error");
+
+      this.props.updateStatusCallback(error.message, 'error')
+
+    });
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.boardName && this.props.boardName !== prevProps.boardName) {
+      this.getBoard(Cards_URL, this.props.boardName)
```

Comment:
It seems like the use of this `componentDidUpdate()` lifecycle method is to make sure the board gets "reset" when a different board is selected in the dropdown list.